src/shelly_switch.cpp

```diff
@@ -316,4 +334,46 @@ void ShellySwitch::InputEventHandler(Input::Event ev, bool state) {
   }
 }
 
+void ShellySwitch::AddPowerMeter(uint16_t *iid) {
+  if (out_pm_ == nullptr) return;
+
+  // Power
+  power_char_ = new mgos::hap::UInt16Characteristic(
+      (*iid)++, &kHAPCharacteristic_EveConsumption, 0, 65535, 1,
+      [this](HAPAccessoryServerRef *,
+             const HAPUInt16CharacteristicReadRequest *, uint16_t *value) {
+        *value = out_pm_->GetPowerW().ValueOrDie();
+        return kHAPError_None;
+      },
+      true /* supports_notification */, nullptr, "eve-power-consumption");
+  AddChar(power_char_);
+  // Energy
+  total_power_char__ = new mgos::hap::UInt16Characteristic(
+      (*iid)++, &kHAPCharacteristic_EveTotalConsumption, 0, 65535, 1,
+      [this](HAPAccessoryServerRef *,
+             const HAPUInt16CharacteristicReadRequest *, uint16_t *value) {
+        *value = out_pm_->GetEnergyWH().ValueOrDie() / 1000.0f;
+        return kHAPError_None;
+      },
+      true /* supports_notification */, nullptr, "eve-total-power-consumption");
+  AddChar(total_power_char__);
+
+  power_timer_.Reset(1000, MGOS_TIMER_REPEAT);
+}
+
+void ShellySwitch::PowerMeterTimerCB() {
+  float current_power = out_pm_->GetPowerW().ValueOrDie();
```

Comment:
careful there, you don't want to die just because power meter had a hiccup.
always check if returned value is ok and take evasive action.
no dying please :)

Reply:
wdym? it's never ok to call ValueOrDie without checking for .ok() first.
i.e. you must always make sure the program won't actually die, it's never of for firmware to keel over and abort.
"OrDie" in the name is there to remind you of that.